src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,188 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    private class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        private Node(Node<T> prev, T element, Node<T> next) {
+            this.value = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (first == null) {
+            linkFirst(value);
+            return;
+        }
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == 0) {
+            linkFirst(value);
+            return;
+        }
+        if (index == size) {
+            linkLast(value);
+            return;
+        } else {
+            linkBefore(value, getNode(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        Object[] listToAdd = list.toArray();
+        int countToAdd = listToAdd.length;
+        for (int i = 0; i < countToAdd; i++) {
+            add((T) listToAdd[i]);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> nodeToSet = getNode(index);
+        T oldVal = nodeToSet.value;
+        nodeToSet.value = value;
+        return oldVal;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkElementIndex(index);
+        return unlink(getNode(index));
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> x = first; x != null; x = x.next) {
+            if (object == null) {
+                if (x.value == null) {
+                    unlink(x);
+                    return true;
+                }
+                continue;
+            }
+            if (object.equals(x.value)) {
+                unlink(x);
+                return true;
+            }
+        }
         return false;
```

Comment:
```suggestion
        for (Node<T> x = first; x != null; x = x.next) {
            if (object == x.value || object != null && object.equals(x.value)) {
                    unlink(x);
                    return true;
            }
        }
        return false;
```

Reply:
and i make this mistake once more time))